packages/api/src/derives/fees/estimateFee.ts

```diff
@@ -0,0 +1,56 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import Extrinsic from '@cennznet/types/extrinsic/Extrinsic';
+import {generateTransactionPayment} from '@cennznet/types/runtime/transaction-payment/TransactionPayment';
+import {AnyAssetId, IExtrinsic} from '@cennznet/types/types';
+import {drr} from '@polkadot/rpc-core/rxjs';
+import {TypeRegistry} from '@polkadot/types';
+import {RuntimeDispatchInfo} from '@polkadot/types/interfaces/rpc';
+import ExtrinsicEra from '@polkadot/types/primitive/Extrinsic/ExtrinsicEra';
+import BN from 'bn.js';
+import {combineLatest, Observable, of} from 'rxjs';
+import {catchError, first, map, switchMap} from 'rxjs/operators';
+
+export function estimateFee(api: ApiInterfaceRx) {
```

Comment:
Would suggest add some comment to describe what this interface is about, and how and when it can be used, etc. So that our API consumers do not need to look through the code and tests to grab the whole idea. 

If possible, it is even better to describe what is the currency of `estimateFee`, with giving some examples. And we eventually need to get it documented somewhere.

Reply:
Have added these lines,
// We generate fake signature data here to ensure the estimated fee will correctly match the fee paid when the extrinsic is signed by a user.
   // This is because fees are currently based on the byte length of the extrinsic

Reply:
May be I can add, this interface is do determine fee estimate for any transaction that is going to be executed.. The estimate can be in any currency provided there is enough liquidity in the exchange pool for that currency.
Do you think this is enough?

Reply:
A simple code example or a scenario case would explain more clearly, so that developers do not need to check test cases to see how to use it. Maybe we could add that in the future, no more struggling here if you think it is unnecessary :)